cmake/machinecfg/Darwin.cmake

```diff
@@ -0,0 +1,157 @@
+        directory.")
+endif()
+
+if (NOT DEFINED DARWIN_COMPILER)
+    set(DARWIN_COMPILER ${DARWIN_COMPILER_PREFERRED})
+endif()
+
+if (DARWIN_COMPILER STREQUAL "GCC")
+    set(DARWIN_COMPILER ${DARWIN_GCC_PREFERRED})
+endif()
+
+set(DARWIN_PROJECT_PREFIX /projects/parthenon-int/parthenon-project
+    CACHE STRING "Path to parthenon-project checkout")
+set(DARWIN_ARCH_PREFIX ${DARWIN_PROJECT_PREFIX}/views/darwin/${DARWIN_ARCH})
+
+if (NOT EXISTS ${DARWIN_ARCH_PREFIX})
+    message(WARNING "No dependencies detected for \
+        DARWIN_ARCH=\"${DARWIN_ARCH}\" at ${DARWIN_ARCH_PREFIX}")
+    return()
+endif()
+
+string(TOLOWER "${DARWIN_COMPILER}" DARWIN_COMPILER_LOWER)
+set(DARWIN_COMPILER_PREFIX ${DARWIN_ARCH_PREFIX}/${DARWIN_COMPILER_LOWER})
+
+if (NOT EXISTS ${DARWIN_COMPILER_PREFIX})
+    message(WARNING "No dependencies detected for \
+        DARWIN_COMPILER=\"${DARWIN_COMPILER}\" at ${DARWIN_COMPILER_PREFIX}")
+    return()
+endif()
+
+set(DARWIN_VIEW_PREFIX ${DARWIN_COMPILER_PREFIX}/${DARWIN_VIEW_DATE})
+if (NOT EXISTS ${DARWIN_VIEW_PREFIX})
+    message(WARNING "No view detected for \
+        DARWIN_VIEW_DATE=\"${DARWIN_VIEW_DATE}\" at ${DARWIN_VIEW_PREFIX}")
+    return()
+endif()
+
+# Let the user specify the compiler if they really want to. Otherwise, point
+# to the compilers specified by the DARWIN_ options
+if (NOT DEFINED CMAKE_CXX_COMPILER)
+    if (DARWIN_COMPILER MATCHES "GCC")
+        set(GCC_VERSION ${DARWIN_${DARWIN_COMPILER}_VERSION})
+        set(GCC_PREFIX /projects/opt/${DARWIN_ARCH}${DARWIN_MICROARCH_PATH}/gcc/${GCC_VERSION})
+
+        if (DEFINED GCC_VERSION)
+            set(CMAKE_C_COMPILER ${GCC_PREFIX}/bin/gcc
+                CACHE STRING "gcc ${GCC_VERSION}")
+            set(CMAKE_CXX_COMPILER ${GCC_PREFIX}/bin/g++
+                CACHE STRING "gcc ${GCC_VERSION}")
+            set(CMAKE_BUILD_RPATH ${GCC_PREFIX}/lib64
+                CACHE STRING "rpath libs")
+        endif()
+    endif()
+
+    if (NOT DEFINED CMAKE_CXX_COMPILER)
+        message(
+            FATAL_ERROR
+            "Found view on Darwin for compiler version ${DARWIN_COMPILER}, but \
+            don't know how to map it to a specific compiler. Either update \
+            your Parthenon checkout or explicitly set CMAKE_C_COMPILER and \
+            CMAKE_CXX_COMPILER")
+    endif()
+endif()
+
+# MPI - We use the system modules since replicating them in spack can be
+# difficult.
+if (DARWIN_COMPILER MATCHES "GCC")
+    set(MPI_ROOT
+        /projects/opt/${DARWIN_ARCH}${DARWIN_MICROARCH_PATH}/openmpi/${DARWIN_MPI_VERSION}-gcc_${DARWIN_${DARWIN_COMPILER}_VERSION}
+        CACHE STRING "MPI Location")
+endif()
+
+# Kokkos settings
+if (DARWIN_ARCH STREQUAL "x86_64")
+    set(Kokkos_ARCH_HSW ON CACHE BOOL "Target Haswell")
+elseif(DARWIN_ARCH STREQUAL "ppc64le")
+    # set(Kokkos_ENABLE_CUDA ON CACHE BOOL "Cuda")
+    set(Kokkos_ARCH_POWER9 ON CACHE BOOL "Target Power9")
+    # set(Kokkos_ARCH_VOLTA70 ON CACHE BOOL "Target V100s")
```

Comment:
Are these commented out lines needed?

Reply:
Also if by default we are going to be running with 2 mpi procs, it might be worth setting. 
```cmake
set(NUM_GPU_DEVICES_PER_NODE "2" CACHE STRING "Number of gpu devices to use when testing if built with Kokkos_ENABLE_CUDA")
```

The default is a single gpu.

Reply:
Yeah, I'm working on re-enabling these right now.

Good suggestion.

Reply:
Oh, I need to fix this for the x86 volta nodes - they only have 1 GPU.